retis-events/src/ovs.rs

```diff
@@ -28,41 +28,59 @@ pub enum OvsEventType {
     /// Upcall event. It indicates the begining of an upcall. An upcall can have multiple enqueue
     /// events.
     #[serde(rename = "upcall")]
-    Upcall(UpcallEvent),
+    Upcall {
+        #[serde(flatten)]
+        upcall: UpcallEvent,
+    },
 
     /// Upcall enqueue event. It indicates a packet (fragment) is enqueued for userspace
     /// processing.
     #[serde(rename = "upcall_enqueue")]
-    UpcallEnqueue(UpcallEnqueueEvent),
+    UpcallEnqueue {
+        #[serde(flatten)]
+        upcall_enqueue: UpcallEnqueueEvent,
+    },
 
     /// Upcall return event. It indicates an upcall has ended.
     #[serde(rename = "upcall_return")]
-    UpcallReturn(UpcallReturnEvent),
+    UpcallReturn {
+        #[serde(flatten)]
+        upcall_return: UpcallReturnEvent,
+    },
 
     /// Receive upcall event. It indicates userspace has received an upcall.
     #[serde(rename = "recv_upcall")]
-    RecvUpcall(RecvUpcallEvent),
+    RecvUpcall {
+        #[serde(flatten)]
+        recv_upcall: RecvUpcallEvent,
+    },
 
     /// Operation event. It indicates userspace has executed a flow operation on an upcalled
     /// packet.
     #[serde(rename = "flow_operation")]
-    Operation(OperationEvent),
+    Operation {
+        #[serde(flatten)]
+        flow_operation: OperationEvent,
+    },
 
     /// Action execution event. It indicates the datapath has executed an action on a packet.
     #[serde(rename = "action_execute")]
-    Action(ActionEvent),
+    Action {
+        #[serde(flatten)]
+        action_execute: ActionEvent,
+    },
 }
 
 impl EventFmt for OvsEventType {
```

Comment:
Testing this change, I found it very inconvenient that, at the python level, you have to do:
```
>> In [6]: e["ovs"]                                                                                                                                                                                                                            
Out[6]: OvsEvent { event: Upcall { upcall: UpcallEvent { cmd: 1, port: 2271867293, cpu: 2 } } }                                                                                                                                             
                                                                                                                                                                                                                                            
In [7]: e["ovs"].event                                                                                                                                                                                                                      
Out[7]: Upcall { upcall: UpcallEvent { cmd: 1, port: 2271867293, cpu: 2 } }                                                                                                                                                                 
                                                                                                                                                                                                                                            
In [8]: e["ovs"].upcall      
```

So I quickly wrote #458 . I think it's worth adding it to this series or at least to 1.5. WDYT?

Reply:
That makes perfect sense and IMO is a good material for 1.5. Just checking: this does not change the JSON representation?

Reply:
BTW I can cherry-pick it here if there's a new version needed.

Reply:
It is now included in this PR.